src/BitNode/ui/BitNodeAdvancedOptions.tsx

```diff
@@ -0,0 +1,213 @@
+import { type BitNodeBooleanOptions } from "@nsdefs";
+import React from "react";
+import { Box, ButtonGroup, Collapse, Paper, TextField, Typography } from "@mui/material";
+import { OptionSwitch } from "../../ui/React/OptionSwitch";
+import { ButtonWithTooltip } from "../../ui/Components/ButtonWithTooltip";
+
+interface SourceFileButtonRowProps {
+  sourceFileNumber: number;
+  sourceFileLevel: number;
+  callbacks: BitNodeAdvancedOptionsProps["callbacks"];
+}
+
+function SourceFileButtonRow({
+  sourceFileNumber,
+  sourceFileLevel,
+  callbacks,
+}: SourceFileButtonRowProps): React.ReactElement {
+  const [sfLevel, setSfLevel] = React.useState<number>(sourceFileLevel);
+  const [inputValue, setInputValue] = React.useState<string>(sourceFileLevel.toString());
+
+  const title = `SF-${sourceFileNumber}`;
+  const sourceFileLevelTool =
+    sourceFileNumber !== 12 ? (
+      Array.from([0, 1, 2, 3]).map((level) => (
```

Comment:
Shouldn't this be using SourceFileInfo or something?

Reply:
I don't understand your question. Can you clarify it?

Reply:
I meant instead of hardcoding that everything has 3 SF levels, shouldn't we be getting that from the data map we've created?

Reply:
I do it like that to make all SF have the same number of buttons (except SF12). All SF have 3 buttons, but we only enable the relevant ones (with data in the map).